var/spack/repos/builtin/packages/netcdf/package.py

```diff
@@ -47,6 +48,7 @@ class Netcdf(Package):
 
     # Required for NetCDF-4 support
     depends_on("zlib")
+    depends_on("hdf5@:1.8.999", when='+hdf5_18')
```

Comment:
@davydden I don't know. Last time I tried this with another package (Python) it didn't work; I had to write `:2.7.999` there.

I wonder what `:1.8` is supposed to mean. Doesn't it mean "up to and not including version 1.8"? The idea here is to avoid version 1.10, not version 1.8.